sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/permissions/PermissionRequestActivity.java

```diff
@@ -0,0 +1,139 @@
+
+    /**
+     * Start activity for requesting permissions.
+     *
+     * @param context The context from which the activity will be started.
+     * @param permissions List of requested permissions.
+     * @return Future with the result of a permissions request.
+     */
+    public static AppCenterFuture<Result> requestPermissions(Context context, String... permissions) {
+        if (sResultFuture != null) {
+            AppCenterLog.error(LOG_TAG, "Result future flag is null.");
+            return null;
+        }
+        sResultFuture = new DefaultAppCenterFuture<>();
+        Intent intent = new Intent(context, PermissionRequestActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+        intent.putExtra(EXTRA_PERMISSIONS, permissions);
+        context.startActivity(intent);
+        return sResultFuture;
+    }
+
+    @VisibleForTesting
+    static void complete(Result result) {
+        if (sResultFuture != null) {
+            sResultFuture.complete(result);
+            sResultFuture = null;
+            return;
+        }
+        AppCenterLog.debug(LOG_TAG, "The start of the activity was not called using the requestPermissions function or the future has already been completed");
+    }
+
+    @Nullable
+    private String[] getPermissionsList() {
+        Intent intent = getIntent();
+        if (intent == null) {
+            return null;
+        }
+        Bundle extras = intent.getExtras();
+        if (extras == null) {
+            return null;
+        }
+        return extras.getStringArray(EXTRA_PERMISSIONS);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            Exception exception = new Exception("There is no need to request permissions in runtime on Android earlier than 6.0.");
+            AppCenterLog.error(LOG_TAG, "Android version incompatible.", exception);
+            complete(new Result(false, exception));
+            finish();
+            return;
+        }
+        String[] permissions = getPermissionsList();
+        if (permissions == null) {
+            Exception exception = new Exception("Failed to get permissions list from intents extras.");
+            AppCenterLog.error(LOG_TAG, "Failed to get permissions list.", exception);
+            complete(new Result(false, exception));
+            finish();
+            return;
+        }
+        requestPermissions(permissions, REQUEST_CODE);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        if (requestCode == REQUEST_CODE) {
+            boolean isGranted = grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
```

Comment:
It's NOT valid in general case. You're receiving multiple permissions in argument, but result is just first one